### PR TITLE
fix #6587 feat(nimbus): support multiple feature configs

### LINKED ISSUE #6587: Support multiple features/values in branch model
We need to change the NimbusExperiment/Branch models to support multiple features/values.  Let's change the models, update the tests, and modify the serializers without exposing multiple values through the serializers.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-1841)

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* Experiments need to be able to target multiple features
* We'll need to store multiple FeatureConfigs on an experiment
* We'll need to store multiple feature values on each branch

This commit

* Changes the foreign key on NimbusExperiment to a many to many field
* Adds a new model to store a feature value for each feature config on each branch
* Updates the NimbusExperimentFactory to populate the data correctly
* Shims the V5 and V6 APIs to maintain backwards compatibility with the existing interfaces
* Data migrates from the old structure to the new one

### CODE REVIEW

**Review by lmorchard — CHANGES_REQUESTED** [review]
I think I caught a couple of things:

1. The screenshot cloning seems redundant.

1. The integration test failure looks legit. When I create an experiment locally, I get a "field may not be blank" error from GQL for a disabled feature value when I hit "Save and Continue". But, since the feature is disabled, the value field is not revealed to edit or display the validation error. That looks like what the integration test is tripping over

image

File: app/experimenter/experiments/models/nimbus.py [hunk 1]
```diff
@@ -588,6 +590,16 @@ def clone(self, to_experiment):
...
 
+        for feature_value in self.feature_values.all():
+            feature_value.id = None
+            feature_value.branch = cloned
+            feature_value.save()
+
+        for screenshot in self.screenshots.all():
+            screenshot.id = None
+            screenshot.branch = cloned
+            screenshot.save()
```
[the remark this answers is not in the document]
Comment (jaredlockhart) [reply app/experimenter/experiments/models/nimbus.py:601]: Oh you know what this is a bad rebase, I'll take that out.

**Comment by jaredlockhart** [comment]
Yep I gotta fix that form, good job integration tests!

**Review by lmorchard — APPROVED** [review]
Looks fixed up to me!

**Pull request merged**